plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/preferences/RunOptionsDefaultsComponent.java

```diff
@@ -233,93 +330,105 @@ public void setEnabled(boolean enabled) {
     stagingLocationInput.setEnabled(enabled);
   }
 
+
+  /**
+   * Return the fetched staging locations result future, or null if no job is yet scheduled.
+   */
+  private synchronized Future<SortedSet<String>> getStagingLocationsResult() {
```

Comment:
This method is called only in the UI thread, so `synchronized` is not necessary. Please see my comment below.